public/javascripts/tournament.js

```diff
@@ -9,34 +16,175 @@ function getLastPartOfUrl() {
+            availableMatches.push(match);
+        }
+    });
+    return availableMatches;
+}
+
+function getPlayerDictionary(players) {
+    var playerDictionary = {};
+    players.forEach(function(player) {
+        playerDictionary[player.participant.id] = player.participant.name || player.participant.username;
+    });
+    return playerDictionary;
+}
+
+function getMatchDictionary(matches) {
+    var matchDictionary = {};
+    matches.forEach(function(match) {
+        matchDictionary[match.match.id] = match.match;
+    });
+    return matchDictionary;
+}
+
+function buildSetupSelector(setupList) {
+	var $selector = $('<select class="setup-selector"><option value="" disabled selected style="display:none;"></option></select>');
+	for (var i = 0; i < setupList.length; i++) {
+		if (!setupList[i].status || setupList[i].status === 'Open') {
+			$selector.append('<option value=' + (i + 1) + '>' + (i + 1) + '</option>');
+		}
+	}
+	return $selector;
+}
+
+function updateMatchList(availableMatches, players) {
+	var $matchList = $('#matches-tbody');
+	var $setupSelector = buildSetupSelector(setups);
+	$matchList.empty();
+	availableMatches.forEach(function(match) {
+		var tableRow = [];
+		var player1Id = match.match.player1_id;
+		var player2Id = match.match.player2_id;
+		var player1 = players[player1Id];
+		var player2 = players[player2Id];
+		tableRow.push('<tr class="available-match-row" data-id="' + match.match.id + '"><td>');
+		tableRow.push(player1 + ' vs. ' + player2);
+		tableRow.push('</td><td>');
+		tableRow.push('<div class="setup-selector-wrap"></div>');
+		tableRow.push('</td></tr>');
+		$matchList.append(tableRow.join(''));
+	});
+	$('.setup-selector-wrap').append($setupSelector.clone());
+}
+
+function updateSetupsList(setupsList) {
+	var $setupsList = $('#setups-list');
+	$setupsList.empty();
+	for (var i = 0; i < setupsList.length; i++) {
+		var setup = setupsList[i];
+		var $setup = $('<li class="setup-row" data-id="' + (i + 1) + '"></li>');
+		$setup.append('<div class="setup-id">' + (i + 1) + '</div>');
+		if (setup.status) {
+			$setup.append('<div class="setup-status">' + setup.status + '</div>');
+		}
+		if (setup.status === 'Assigned') {
+			var match = matchDict[setup.matchId];
+			if (match) {
+				if (match.state === 'open') {
+					var player1 = playerDict[match.player1_id];
+					var player2 = playerDict[match.player2_id];
+					$setup.append('<div>' + player1 + ' vs ' + player2 + '</div>');
+				} else {
+					tournamentSocket.emit('open setup', i);
+				}
+			}
+			$setup.append('<a href="" class="reassign-setup">Reassign</a>');
+		}
+		$setupsList.append($setup);
+	}
+}
+
 $(function() {
+	$(document).on('change', '.setup-selector', function() {
+		var setupIndex = $(this).val() - 1;
+		var tableRow = $(this).closest('.available-match-row');
+		var matchId = tableRow.data('id');
+		tournamentSocket.emit('assign setup', setupIndex, matchId);
+	});
+
+	$(document).on('click', '.reassign-setup', function(e) {
```

Comment:
No, `.click()` is just shorthand for `.on('click'`. `.on('click'` allows for event delegation, like I use here. It's probably better to use a similar style everywhere, so I changed the `.click` handlers.